### ISSUE TITLE
Why does Discriminator return negative value?

### ISSUE THREAD

**Issue by QsROg8320** [NVlabs/stylegan3#94]
Why does Discriminator return negative value? I think that Discriminator should return a value between 0 and 1.
```Python
torch.cuda.empty_cache()
script_dir = os.path.abspath('') #<-- absolute dir the script is in
rel_path = 'stylegan3-r-afhqv2-512x512.pkl'
abs_file_path = os.path.join(script_dir, rel_path)
with open(abs_file_path, 'rb') as f:
    D = pickle.load(f)['D']  # torch.nn.Module
with open(abs_file_path, 'rb') as f:
    G = pickle.load(f)['G_ema'] 
z = torch.randn([1, G.z_dim])  # latent codes
c = None                                # class labels (not used in this example)
img = G(z, c)                           # NCHW, float32, dynamic range [-1, +1], no truncation

d_out = D(img,c)
print(d_out)
#tensor([[-2.0559]])

```

**Comment by PDillis** [comment]
The Discriminator's output is not a probability/classification of `real` or `fake` ever since [Progressive Growing of GANs](https://arxiv.org/abs/1710.10196); I recommend you read about [Wasserstein GAN](https://arxiv.org/abs/1701.07875) as it's the basis of most modern GANs, but it does exactly what you are asking: the [output layer is a linear/fully connected layer](https://github.com/NVlabs/stylegan3/blob/a5a69f58294509598714d1e88c9646c3d7c6ec94/training/networks_stylegan2.py#L698), which can give a real number. A summary on WGAN loss can be found [here](https://machinelearningmastery.com/how-to-implement-wasserstein-loss-for-generative-adversarial-networks/).
